core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -510,8 +510,19 @@ class SparkSubmitSuite
         "my.great.lib.MyLib", "my.great.dep.MyLib")
 
       val appArgs = new SparkSubmitArguments(clArgs)
-      val (_, _, sparkConf, _) = submit.prepareSubmitEnvironment(appArgs)
-      sparkConf.get("spark.jars").contains("mylib") shouldBe true
+      try {
+        val (_, _, sparkConf, _) = submit.prepareSubmitEnvironment(appArgs)
+        sparkConf.get("spark.jars").contains("mylib") shouldBe true
+      } finally {
+        val mainJarPath = Paths.get("my.great.dep_mylib-0.1.jar")
+        val depJarPath = Paths.get("my.great.lib_mylib-0.1.jar")
+        if (Files.exists(mainJarPath)) {
+          Files.delete(mainJarPath)
+        }
+        if (Files.exists(depJarPath)) {
+          Files.delete(depJarPath)
```

Comment:
isn't this removed at the JVM shutdown hook?

Reply:
1.yeah, it was not cleared when the jvm was shut down.

2.The logic it generates is here:
https://github.com/apache/spark/blob/be95298f24669cddaa65e412d7681476149e2977/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L452-L454

3.We can simply reproduce it by any of the following commands:
```
- ./build/sbt "core/testOnly *SparkSubmitSuite*"
- ./build/sbt "core/testOnly *SparkSubmitSuite -- -z \"SPARK-35084: include jars of the --packages in k8s client mode & driver runs inside a POD\""
```

Reply:
ah okie. LGTM